rstudio/rstudio
# Fix datatable column number and paging bug

### Intent

Addresses https://github.com/rstudio/rstudio/issues/9036

### Approach

The callback for number of columns returns the offset and number of shown columns. We were just printing out those numbers raw instead of adding them for showing column numbers.

I also found an exception when paging around quickly. This may have been exacerbated by the recent jQuery update due to ordering/performance changes.

### Automated Tests

There is no infrastructure for datatable automation at the moment and building it during Juliet Rose is out of scope.

### QA Notes

Ensure that the number of columns updates properly when paging around. Also check that no exceptions are thrown in the console when doing said paging.

### Checklist

- [x ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ x] This PR passes all local unit tests
